homeassistant/components/system_bridge/coordinator.py

```diff
@@ -188,7 +152,10 @@ async def _setup_websocket(self) -> None:
                     session=async_get_clientsession(self.hass),
                 )
 
-            self.hass.async_create_task(self._listen_for_data())
+            self.hass.async_create_background_task(
+                self.websocket_client.listen(callback=self.async_handle_module),
+                name="System Bridge WebSocket Listener",
+            )
```

Comment:
```suggestion
            self.hass.async_create_background_task(
                self._listen_for_data(),
                name="System Bridge WebSocket Listener",
            )
```

I think you could keep `_listen_for_data ` and make the whole thing a background task instead of removing the logic to trap exceptions since if `self.websocket_client.listen` raises (which it clearly can) none of the exceptions are going to be handled and we will get a warning about unhandled exceptions

Reply:
Good point